Clinical-Genomics/scout
# Support for private IGV tracks and alignment files from Amazon cloud

Adds the support to load private cloud s3 storage objects (alignments, tracks) into the IGV track of Scout.

![image](https://user-images.githubusercontent.com/28093618/77896235-3e97d000-7278-11ea-92fb-d2051e52d015.png)


Tested so far with Cosmic tracks stored on Amazon AWS S3 storage. Works like this:
- Connects to the cloud using an aws_access_key_id and aws_secret_access_key and generates a temporary link (presigned url) to the resource. The link expires after a few minutes (500 seconds tentatively). An endpoint in the alignviewers views (cloud_resource) acts as a proxy returning the requested resource. This way the presigned_url (a https link) is never visualized in the pages, and the resource is secure (can't be downloaded using the link)

**How to test**:
1. Add the following parameters to the scout stage config file:
```
# Cloud storage resources accession params (i.e. IGV tracks)
REGION_NAME = "eu-north-1"
ACCESS_KEY = "VALID_ACCESS_KEY"
SECRET_ACCESS_KEY = "VALID_SECRET_ACCESS_KEY"
BUCKET_NAME = "AWS_BUCKET_NAME"
FOLDER = "bucket_folder"
```
contact me (@northwestwitch ) to obtain valid coordinates

Test on a local instance of scout
1. load the cancer demo case (not really, I've made it a cancer case because it has MT bam files):
`scout --demo load scout/demo/643595.config.yaml`
1. Go to a MT variant (alignment files available):
http://localhost:5000/cust000/643595/28d11fd775234aeee4fed545420f8fd1
1. Click on the igv viewer button and you'll open the igv page
1. Using the igv viewer search box go to some gene not on MT chr, and make sure that the cosmic tracks are displayed correctly.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] tests executed by CR (locally)
